src/Integration.UnitTests/Binding/RoslynProjectBinderTests.cs

```diff
@@ -0,0 +1,213 @@
+
+            fileSystemMock.Setup(x => x.File.Exists(mockRulesetPath)).Returns(true);
+            ruleSetSerializerMock.Setup(x => x.LoadRuleSet(mockRulesetPath)).Returns(null as RuleSet);
+
+            var result = testSubject.IsBound(bindingConfiguration, projectMock);
+            result.Should().BeFalse();
+        }
+
+        [TestMethod]
+        public void IsBound_RulesetFileHasNoDeclarations_False()
+        {
+            var bindingConfiguration = GetBindingConfiguration();
+            var projectMock = GetCSharpProject();
+            var mockRulesetPath = SetupRulesetPath();
+
+            fileSystemMock.Setup(x => x.File.Exists(mockRulesetPath)).Returns(true);
+            
+            ruleSetSerializerMock.Setup(x => x.LoadRuleSet(mockRulesetPath)).Returns(new RuleSet("test"));
+
+            solutionRuleSetsInformationProviderMock
+                .Setup(x => x.GetProjectRuleSetsDeclarations(projectMock))
+                .Returns(Array.Empty<RuleSetDeclaration>());
+
+            var result = testSubject.IsBound(bindingConfiguration, projectMock);
+            result.Should().BeFalse();
+        }
+
+        [TestMethod]
+        public void IsBound_RulesetFileHasOneDeclaration_CantLoadProjectRulesetFile_False()
+        {
+            var bindingConfiguration = GetBindingConfiguration();
+            var projectMock = GetCSharpProject();
+            var mockRulesetPath = SetupRulesetPath();
+
+            fileSystemMock.Setup(x => x.File.Exists(mockRulesetPath)).Returns(true);
+
+            ruleSetSerializerMock.Setup(x => x.LoadRuleSet(mockRulesetPath)).Returns(new RuleSet("test"));
+
+            var ruleSetDeclaration = GetRuleSetDeclaration(projectMock);
+
+            solutionRuleSetsInformationProviderMock
+                .Setup(x => x.GetProjectRuleSetsDeclarations(projectMock))
+                .Returns(new List<RuleSetDeclaration> {ruleSetDeclaration});
+
+            var filePath = "";
+            solutionRuleSetsInformationProviderMock
+                .Setup(x => x.TryGetProjectRuleSetFilePath(projectMock, ruleSetDeclaration, out filePath))
+                .Returns(false);
+
+            var result = testSubject.IsBound(bindingConfiguration, projectMock);
+            result.Should().BeFalse();
+        }
+
+        private RuleSetDeclaration GetRuleSetDeclaration(ProjectMock projectMock)
+        {
+            var mockDeclaration =
+                new RuleSetDeclaration(projectMock, new PropertyMock("name", null), "test path", null);
+            
+            return mockDeclaration;
+        }
+
+        private string SetupRulesetPath()
+        {
+            var mockRulesetPath = "c:\\test.ruleset";
+
+            solutionRuleSetsInformationProviderMock
+                .Setup(x => x.CalculateSolutionSonarQubeRuleSetFilePath("key", Language.CSharp, SonarLintMode.Connected))
+                .Returns(mockRulesetPath);
+
+            return mockRulesetPath;
```

Comment:
There isn't a test for `IsBound` returning `true` (so you could simplifying the implementation to hard-code a return value of `false` and the tests would all still pass ;-) )

Reply:
I know, this is because the RuleSetIncludeChecker is a static class :( I wanted to minimize the scope of the commit

Reply:
Ok; if it's deliberate then call it out in a review comment, and publicly record that it's still work to be done (e.g. any empty placeholder test that does nothing)

Reply:
Good idea 👍 will do